DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py

```diff
@@ -0,0 +1,137 @@
+import asyncio
+import pytest
+
+from dent_os_testbed.lib.bridge.bridge_vlan import BridgeVlan
+from dent_os_testbed.lib.ip.ip_link import IpLink
+from dent_os_testbed.utils.test_utils.tgen_utils import (
+    tgen_utils_get_dent_devices_with_tgen,
+    tgen_utils_setup_streams, tgen_utils_start_traffic,
+    tgen_utils_stop_traffic, tgen_utils_get_traffic_stats,
+    tgen_utils_get_loss, tgen_utils_dev_groups_from_config,
+    tgen_utils_traffic_generator_connect, tgen_utils_stop_protocols)
+
+pytestmark = pytest.mark.suite_functional_vlan
+
+packet_vids = ["X", 1, 2, 3, 4]  # VLAN tag number contained in the transmitted packet
+port_map = (
+    {"port": 0, "settings": [{"vlan": 2, "untagged": False, "pvid": False},
+                             {"vlan": 3, "untagged": True, "pvid": False},
+                             {"vlan": 4, "untagged": True, "pvid": True}]},
+    {"port": 1, "settings": [{"vlan": 2, "untagged": False, "pvid": False}]},
+    {"port": 2, "settings": [{"vlan": 3, "untagged": True, "pvid": True}]},
+    {"port": 3, "settings": [{"vlan": 4, "untagged": True, "pvid": True}]},
+)
```

Comment:
What do you think about using a dict instead of a tuple for the port map? If you move the map inside of the test function then you can use the ports` names as keys:
```python
port_map = {
    ports[0]: [ {"vlan": 2, "untagged": False, "pvid": False}, ... ],
    ports[1]: [ { ... } ],
    ...
}
```

Reply:
this config of ports VLAN will be applied to 2 more tests in file:` test_vlan_all_supported_modes_multicast, test_vlan_all_supported_modes_unicast`